pandas/tests/dtypes/test_common.py

```diff
@@ -706,3 +707,13 @@ def test__get_dtype_fails(input_param, expected_error_message):
 )
 def test__is_dtype_type(input_param, result):
     assert com._is_dtype_type(input_param, lambda tipo: tipo == result)
+
+
+@pytest.mark.parametrize("val", [np.datetime64("NaT"), np.timedelta64("NaT")])
```

Comment:
can you add an additional test for when dtype == object (which is the other path that hits M8 and m8)

Reply:
This would be a test separate from `test_astype_nansafe` to check that `NaT` is still null after casting to `object`?

Reply:
yep